src/main/java/battlestates/states/UserTurnState.java

```diff
@@ -0,0 +1,145 @@
+package battlestates.states;
+
+import core.ChoiceInputValidator;
+import interfaces.State;
+import io.InputValidator;
+import io.Output;
+import io.OutputHandler;
+import objects.battle.Skill;
+import objects.battle.PlayerSkillHandler;
+import objects.battle.StatDisplayer;
+import objects.battle.enemy.SkillHelper;
+import objects.character.EnemyFacade;
+import objects.character.Player;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class UserTurnState implements State {
+    /**
+     * State which probes and determines what the user does on their turn during a battle.
+     * Attributes:
+     * user: Player object representing the user engaged in the battle
+     * foe: EnemyFacade object that the user is engaged in battle with
+     * done: Boolean showing whether the state is done
+     * skillList: String representation of the skills in user
+     * awaitingInp: Boolean showing whether the state is waiting for input
+     */
+    private Player user;
+    private EnemyFacade foe;
+    private boolean done = false;
+    private List<String> skillList;
+    private boolean awaitingInp = false;
+    private InputValidator validator;
+    private int questionNum = 0;
+    private final OutputHandler output = Output.getScreen();
+    private final List<String> menuList = Arrays.asList("Skills", "Inventory", "Stats");
+
+    public UserTurnState(Player user, EnemyFacade foe) {
+        // Will need to change later to accommodate other options, like inventory
+        SkillHelper dummy = new SkillHelper();
+        this.skillList = dummy.toSkillString(user.getSkillList());
+        this.skillList.add("Back");
+        this.user = user;
+        this.foe = foe;
+    }
+
+    /**
+     * Asks the user for input on what action they want to do.
+     */
+    @Override
+    public void preInput() {
+        switch (questionNum) {
+            case 0:
+                // Asking the user for input
+                output.generateTextWithOptions("Select an option:", menuList);
+                this.validator = new ChoiceInputValidator(menuList);
+                break;
+            case 1:
+                output.generateTextWithOptions("Pick a skill:", skillList);
```

Comment:
As brain said, this should go to different files so it does not violate the single responsibility principle.